crates/cubecl-reduce/src/args.rs

```diff
@@ -0,0 +1,332 @@
+use std::marker::PhantomData;
+
+use cubecl::prelude::*;
+use cubecl_core::{self as cubecl, unexpanded};
+use cubecl_std::tensor::r#virtual::{
+    ReadWrite, VirtualTensor, VirtualTensorOperations, VirtualTensorOperationsExpand,
+};
+
+pub trait ReducePrecision {
+    type In: Numeric;
+    type Out: Numeric;
+}
+
+impl<In: Numeric, Out: Numeric> ReducePrecision for (In, Out) {
+    type In = In;
+    type Out = Out;
+}
```

Comment:
I am not a big fan of the name `Precision`. Doesn't really make sense when `In` or `Out` are integers. I suggest `ReduceDType` or `ReduceIO` instead?